shims/spark330/src/main/scala/com/nvidia/spark/rapids/shims/spark330/Spark330Shims.scala

```diff
@@ -16,11 +16,18 @@
 
 package com.nvidia.spark.rapids.shims.spark330
 
+import ai.rapids.cudf.ColumnVector
 import com.nvidia.spark.rapids._
 import com.nvidia.spark.rapids.shims.v2._
 
 class Spark330Shims extends Spark33XShims {
   override def getSparkShimVersion: ShimVersion = SparkShimServiceProvider.VERSION
 
   override def isCastingStringToNegDecimalScaleSupported: Boolean = true
+
+  override def throwIndexOutOfBoundsException(
```

Comment:
We have been moving away from the v1 shims and preferring to use the v2 shims.  This PR already has an object class to hold the logic for throwing the right exception, so we simply need to move that object class into `com.nvidia.spark.rapids.shims.v2` and use that directly at compile-time rather than a runtime lookup of the shim.